metrics/web/lib/common/data/repositories/firestore_build_day_repository.dart

```diff
@@ -0,0 +1,60 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:metrics/common/data/model/build_day_data.dart';
+import 'package:metrics/common/domain/entities/build_day.dart';
+import 'package:metrics/common/domain/entities/persistent_store_error_code.dart';
+import 'package:metrics/common/domain/entities/persistent_store_exception.dart';
+import 'package:metrics/common/domain/repositories/build_day_repository.dart';
+
+/// An implementation of the [BuildDayRepository] for [Firestore].
+class FirestoreBuildDayRepository implements BuildDayRepository {
+  /// A [Firestore] instance this repository uses to interact with the
+  /// Firestore database.
+  final Firestore _firestore = Firestore.instance;
+
+  @override
+  Stream<List<BuildDay>> projectBuildDaysInDateRangeStream(
+    String projectId, {
+    DateTime from,
+    DateTime to,
+  }) {
+    ArgumentError.checkNotNull(projectId, 'projectId');
+
+    final collection = _firestore.collection('build');
+
+    Query query = collection.where('projectId', isEqualTo: projectId);
+
+    if (from != null) {
+      query = query.where('day', isGreaterThanOrEqualTo: from);
+    }
+
+    if (to != null) {
+      query = query.where('day', isLessThanOrEqualTo: to);
+    }
```

Comment:
It seems like the bodies fit. Would it be more readable without blocks? 
![image](https://user-images.githubusercontent.com/70939496/116547966-3dfabe80-a8fc-11eb-9e43-50ef66d17aa2.png)